package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
+	for _script in '/etc/board.d/'*; do
+		if [ ! -s "${_script}" ]; then
+			continue
+		fi
+		# shellcheck source=/dev/null disable=SC2091  # Can't use eval
+		$(. "${_script}")
 	done
-}
+fi
+
+if [ ! -s "${CFG}" ]; then
+	return 1
+fi
 
-[ -s "$CFG" ] || return 1
+return 0
```

Comment:
You can drop that if and final return and simply make the last function statement:
`[ -s "$CFG" ]`

This should yield the desired $? = 0/1 return value.

Reply:
The reason to put this at the end, is to avoid injection attacks, where someone would do `echo 'rm -rf /' >> board_detect`. So best practice is to always add a `exit 0` (or `exit 1`) to the end of your script to ensure they exit when the author intends it to do so. In this case, return, as I don't think we are expected to exit (probably things being sourced potentially? Idk).

The reason the original code was what it was, was probably to try to be `set -e` compliant in the future, as otherwise, just like your statement, would result in an error and cause the script to exit with a failure.